src/org/parosproxy/paros/view/MainMenuBar.java

```diff
@@ -114,10 +128,57 @@ private void initialize() {
 			menuEdit = new javax.swing.JMenu();
 			menuEdit.setText(Constant.messages.getString("menu.edit")); // ZAP: i18n
 			menuEdit.setMnemonic(Constant.messages.getChar("menu.edit.mnemonic"));
+			menuEdit.add(getMenuEditZAPMode());
 		}
 		return menuEdit;
 	}
 
+	private JMenuItem getMenuEditZAPMode(){
+		if (menuZAPMode == null) {
+			menuZAPMode = new JMenu(Constant.messages.getString("menu.edit.zapmode"));
+			menuZAPModeGroup = new ButtonGroup();
+			JRadioButtonMenuItem newButton = null;
+			for (Mode modeType : Mode.values()) {
+				 newButton = addZAPModeMenuItem(modeType);
+  				 menuZAPModeGroup.add(newButton);
+  				 menuZAPMode.add(newButton);
+			}
+			Mode mode = Mode.valueOf(Model.getSingleton().getOptionsParam().getViewParam().getMode());
+			setMode(mode);
+		}
+		return menuZAPMode;
+	}
+
+	private JRadioButtonMenuItem addZAPModeMenuItem(final Mode modeType){
+		final JRadioButtonMenuItem modeItem = new JRadioButtonMenuItem(Constant.messages.getString("view.toolbar.mode."+modeType.name()+".select"));
+		modeItem.addActionListener(new java.awt.event.ActionListener() {
+		@Override
+		public void actionPerformed(java.awt.event.ActionEvent evt) {
+	 		Control.getSingleton().setMode(modeType);
+	 		if (View.isInitialised()){
+	 		    View.getSingleton().getMainFrame().getMainToolbarPanel().setMode(modeType);
+	 		}
+		 }
+		});
+		return modeItem;
+	}
+
+	public void setMode (final Mode mode) {
+		SwingUtilities.invokeLater(new Runnable() {
+			@Override
+			public void run() {
+				Enumeration<AbstractButton> elements = menuZAPModeGroup.getElements();
+				AbstractButton button = null;
+				while (elements.hasMoreElements()) {
+      				   button = elements.nextElement();
+      				   if (button.getText().toLowerCase().contains(mode.name().toString().toLowerCase())) {

```

Comment:
This will not work properly when the modes are translated (already are in some of the languages).